utils/analysis/helpers.go

```diff
@@ -92,6 +92,20 @@ func IsTerminating(run *v1alpha1.AnalysisRun) bool {
 	return false
 }
 
+// GetDryRunMetricNames returns an array of metric names from the Dry-Run array.
+func GetDryRunMetricNames(dryRunArray []v1alpha1.DryRun) (map[string]bool, bool) {
+	metricNames := make(map[string]bool)
+	wildcardPresent := false
+	for _, dryRunObject := range dryRunArray {
+		if dryRunObject.MetricName == "*" {
+			wildcardPresent = true
+			continue
+		}
+		metricNames[dryRunObject.MetricName] = true
+	}
+	return metricNames, wildcardPresent
+}
```

Comment:
I think this method and usage of `GetDryRunMetricNames()` is a bit odd, in that it returns a **_tuple_** of the list of metrics which are supposed to be dry-runned **_or_** a boolean which indicates the user specified wildcard.

To simplify things and make it more generic, I think the method should be changed such that this accepts the `spec.metrics` as a second argument. Then this helper will a subset of `spec.metric` names which are intended to be dry-runned. If the user specified global wildcard `*`, then the returned list of metric names is everything under `spec.metrics`.

Secondly, this would also allow us to support partial wildcards like `foo*bar` in addition to the global wildcard `*`.

Reply:
Great point! I changed it accordingly.